src/rSW_Weather.c

```diff
@@ -838,17 +852,23 @@ SEXP rSW2_calc_SiteClimate(SEXP weatherList, SEXP yearStart, SEXP yearEnd,
 
     allHist = (SW_WEATHER_HIST **)Mem_Malloc(sizeof(SW_WEATHER_HIST *) * numYears,

```

Comment:
In SOILWAT2, I would like to generalize `allocateAllWeather()` so that we can use it here as `allocateAllWeather(&allHist, numYears, &local_LogInfo);` which will replace all of `allHist = ...Mem_Malloc(...); ... init_allHist_years(...); ... for(year = 0; ...) {allHist[year] = ...Mem_Malloc(...); ...`

This includes moving the function `init_allHist_years()` from rSOILWAT2 to SOILWAT2.

Similarly, we get rid of rSOILWAT2's `free_allHist()` and replace it with a generalized `deallocateAllWeather()` so that we can use it as `deallocateAllWeather(&allHist, numYears);`